### PR TITLE
Remove deprecated callbacks from ActionDispatch middlewares

### PR DESCRIPTION
**Pull request by rafaelfranca**
@matthewd I'm most seeking review for the documentation at `ActionDispatch::Reloader` since I'm not sure we want to add more information.

### CODE REVIEW

File: actionpack/lib/action_dispatch/middleware/reloader.rb [hunk 1]
```diff
@@ -1,54 +1,10 @@
...
+  # callbacks, intended to assist with code reloading during development.
   #
   # By default, ActionDispatch::Reloader is included in the middleware stack
   # only in the development environment; specifically, when +config.cache_classes+
-  # is false. Callbacks may be registered even when it is not included in the
-  # middleware stack, but are executed only when <tt>ActionDispatch::Reloader.prepare!</tt>
-  # or <tt>ActionDispatch::Reloader.cleanup!</tt> are called manually.
-  #
+  # is false.
   class Reloader < Executor
```
Comment (kaspth) [inline actionpack/lib/action_dispatch/middleware/reloader.rb]: Curious: why not kill this class?

File: actionpack/lib/action_dispatch/middleware/reloader.rb [hunk 2]
```diff
@@ -1,54 +1,10 @@
...
-  # following:
-  #
-  #     self.response_body = -> (response, output) do
-  #       # code here which refers to application models
-  #     end
-  #
-  # Cleanup callbacks will not be called until after the response_body lambda
-  # is evaluated, ensuring that it can refer to application models and other
-  # classes before they are unloaded.
+  # ActionDispatch::Reloader wraps the request with the ActiveSupport::Reloader
```
Comment (maclover7) [inline actionpack/lib/action_dispatch/middleware/reloader.rb]: small english cleanup:

[...] wraps the request with callbacks provided by `ActiveSupport::Reloader`

File: actionpack/lib/action_dispatch/middleware/reloader.rb (hunk 1 above)
Comment (maclover7) [inline actionpack/lib/action_dispatch/middleware/reloader.rb]: This is an interesting question... if all methods in a class are deprecated, does that mean that the class itself is deprecated? :grimacing:

**Review by maclover7 — CHANGES_REQUESTED** [review]

File: actionpack/lib/action_dispatch/middleware/reloader.rb (hunk 1 above)
Comment (matthewd) [inline actionpack/lib/action_dispatch/middleware/reloader.rb]: Our middleware API assumes it can distinguish entries by their class.

Comment (kaspth) [inline actionpack/lib/action_dispatch/middleware/reloader.rb]: Ah right, `insert_after ActionDispatch::Reloader, SomeMiddleware` and friends 👍

**Comment by rafaelfranca** [comment]
Merged in 3f2b7d60a5

**Pull request closed without merging**